charmcraft/application/main.py

```diff
@@ -40,6 +40,13 @@
     BuildPlannerClass=models.CharmcraftBuildPlanner,
 )
 
+PRIME_BEHAVIOUR_CHANGE_MESSAGE = (
+    "WARNING: the behaviour of the 'prime' keyword has changed in Charmcraft 3. This "
```

Comment:
We should use `CAUTION` to describe potential damage to data or systems. But, the description doesn't make it clear to me that this would have negative effects on data. If we want to sternly notify users of a key point, we should use `IMPORTANT` instead.

Reply:
Well the output charm may be broken, but it wouldn't break their system. Hard to say which one's appropriate?

Reply:
Sounds like a `CAUTION` or `ERROR` is appropriate. FYI, I'm happy to entertain different nomenclature for exceptions/throws ("critical", "error", and so on) than what the style guide calls for, just running off of what I know so far.

I don't fully understand where this error fits into the development workflow, but at a distance it seems to me like we might want to make it clear that this can/will break the charm. Your call. :)